scripts/partition_manager/partition_manager.rst

```diff
@@ -215,6 +214,35 @@ share_size: list
    If none of the partitions in the ``share_size`` list exists, and the partition does not define a ``size`` property, then the partition is removed.
    If none of the partitions in the ``share_size`` list exists, and the partition **does** define a ``size`` property, then the ``size`` property is used to set the size.
 
+All occurrences of a partition name can be replaced with a dict with the key ``one_of``, which is resolved to the first existing partition in the ``one_of`` value.
+It is not allowed to have the ``app`` partition inside such a ``one_of`` dict, because the ``app`` partition always exists.
+
+   .. code-block:: yaml
+      :caption: Example use of a ``one_of`` dict
+
+      # Using 'one_of' in a list like this ...
+      some_span:
+         span: [something, {one_of: [does_not_exist_0, does_not_exist_1, exists1, exists2]}]
+
+      # ... is equivalent to:
+      some_span:
+         span: [something, exists1]
+
+      # Using 'one_of' as a dict value like this ...
+      some_partition:
+         placement:
+            before: {one_of: [does_not_exist_0, does_not_exist_1, exists1, exists2]}
+
+      # ... is equivalent to ...
+      some_partition:
+         placement:
+            before: [exists1]
```

Comment:
Why not convert directly to `exists1` so it's more consistent? If someone wants to wrap it in a list, they can do it themselves.

Reply:
Well, its a bit technical, as the python script will actually convert any non-list items with a one-element-list. So its a bit of lying, but its a very white lie, so I can live with that :D

Reply:
The lines below states this, sort of

Reply:
Go for it! That's the one thing that threw me off a bit.